editor/modes/visual-editor/block.js

```diff
@@ -133,6 +134,10 @@ class VisualEditorBlock extends wp.element.Component {
 
 		const { onSelect, onStartTyping, onHover, onMouseLeave, onFocus, onInsertAfter } = this.props;
 
+		const toolbars = settings.controls && settings.controls.length && ! Array.isArray( first( settings.controls ) )
```

Comment:
Ah, I had fun with this one, but alas, no obvious Lodash-fu will save it 😄 

This is the closest I got, which has a few advantages; handles undefined controls, gracefully handles `controls` of mixed types (instead of testing first entry only), and always returns an array, even if empty, so that we don't need to test truthiness before mapping in the render below.

```js
reduce( settings.controls, ( memo, toolbar ) => (
	Array.isArray( toolbar )
		? [ ...memo, toolbar ]
		: [ ...memo.slice( 0, -1 ), [ ...( last( memo ) || [] ), toolbar ) ] ]
), [] );
```

At the very least, I'm not sure we need `first`, since we can be certain the array exists and has members at that point in the condition. Simply referencing the `[ 0 ]` entry should be enough.